### PR TITLE
Silence compiler warnings on macOS

### PR DESCRIPTION
**Pull request by bradhowes**
* Don't compare unsigned int values as NULL
* Hoist variable declarations to top of function
* Migrate from deprecated identifier

### CODE REVIEW

File: src/drivers/fluid_coreaudio.c [hunk 1]
```diff
@@ -172,6 +176,17 @@ new_fluid_core_audio_driver2(fluid_settings_t *settings, fluid_audio_func_t func
...
+#if MAC_OS_X_VERSION_MIN_REQUIRED < 1060
+    ComponentDescription desc;
+#else
+    AudioComponentDescription desc;
+#endif
+#if MAC_OS_X_VERSION_MIN_REQUIRED < 1060
+    Component comp;
+#else
+    AudioComponent comp;
+#endif
```
[the remark this answers is not in the document]
Comment (bradhowes) [reply src/drivers/fluid_coreaudio.c]: Done

File: src/drivers/fluid_coremidi.c [hunk 2]
```diff
@@ -62,6 +63,10 @@ typedef struct
     int autoconn_inputs;
 } fluid_coremidi_driver_t;
 
+static MIDIClientRef invalid_client = (MIDIClientRef)-1;
+static MIDIEndpointRef invalid_endpoint = (MIDIEndpointRef)-1;
+static MIDIPortRef invalid_port = (MIDIPortRef)-1;
```
[the remark this answers is not in the document]
Comment (bradhowes) [reply src/drivers/fluid_coremidi.c]: Done.

**Comment by bradhowes** [comment]
@derselbst Thanks for the reviews. I checked out the above failures and they all involve pipeline setup failures:

* Timeouts fetching mingw-64 (I was able to manually fetch but it is slow).
* Chocolatey failure due to missing checksums for the pkgconfiglite package -- looks like it can be fixed with `-allow-empty-checksums` flag or manually calculate a checksum and pass it to the Chocolatey command line.

**Review by derselbst — APPROVED** [review]
Yeah, the Windows checks are a bit unstable. The Mac builds are green now, which matter most. Thank you! If you want you can rebase your changes against the `2.2.x` branch. Otherwise I would merge it manually by tomorrow or so, so we can ship this with fluidsynth 2.2.6.

**Comment by derselbst** [comment]
Cherry-picked to 2.2.x and merged back to master. Thanks!

**Pull request closed without merging**